src/Uno.UWP/UI/Core/CoreWindow.iOS.cs

```diff
@@ -1,21 +1,32 @@
-#if __IOS__
 using System;
 using System.Runtime.InteropServices;
 using UIKit;
 using Windows.Foundation;
 using Windows.Foundation.Collections;
 using Windows.Foundation.Metadata;
+using Windows.UI.Core;
 
 namespace Windows.UI.Core
 {
-	public partial class CoreWindow 
+	public partial class CoreWindow  : ICoreWindowEvents
 	{
+		public event TypedEventHandler<CoreWindow, KeyEventArgs> KeyDown;
+
+		public event TypedEventHandler<CoreWindow, KeyEventArgs> KeyUp;
+
         private readonly UIWindow _window;
 
         public CoreWindow(UIWindow window) : this()
         {
             _window = window;
         }
+
+		void ICoreWindowEvents.RaiseKeyDown(KeyEventArgs eventArgs) =>		
```

Comment:
(I guess we don't have logic for key event tunneling for `PreviewKey...`, but bubbling could use same path as pointer events)

Reply:
@MartinZikmund  Catalyst does not support focus at the moment, and there's not even an initial implementation for it.

Reply:
@ajpinedam actually focus is supported everywhere using the ported `FocusManager`, so that should not be a problem 🤔 . Maybe we can discuss on Monday?

Reply:
It could make sense to route those events to the currently focused element in a separate PR.

Reply:
@MartinZikmund, I guess this is what @jeromelaban mentioned in a comment below https://github.com/unoplatform/uno/pull/9288#discussion_r934907149. 

If that's the case, I can add this in another PR. What do you think?